apps/communications/contacts/test/marionette/list_test.js

```diff
@@ -51,6 +51,68 @@ marionette('Contacts > List', function() {
     });
   });
 
+  suite('List Order > ', function() {
+    var data = [['GG', 'E'], ['AA', 'Z'],
+      ['XX', 'C'], ['CC', 'X'], ['EE', 'G'], ['FF', 'F'], ['HH', 'D'],
+      ['BB', 'Y'], ['YY', 'B'], ['ZZ', 'A'], ['DD', 'H']];
+    setup(function() {
+      data.forEach(contactData => {
+        subject.addContact({
+          givenName: contactData[0],
+          familyName: contactData[1]
+        });
+      });
+    });
+
+    function getNamesInList() {
+      return client.executeScript(function() {
+        var names = [];
+        var namesNodes = document.querySelectorAll('li:not([data-group="ice"]).contact-item p');
+        namesNodes = Array.prototype.slice.call(namesNodes);
+        namesNodes.forEach((name) => {
+          names.push(name.textContent)
+        });
+
+        return names;
+      });
+    }
+
+    function givenNameOrder(a, b) {
+      return a[0].localeCompare(b[0]);
+    }
+
+    function familyNameOrder(a, b) {
+      return a[1].localeCompare(b[1]);
+    }
+
+    test('Order by given name > ', function() {
+      var names = getNamesInList();
+      var targetOrder = data.sort(givenNameOrder);
+      names.forEach((name, index) => {
+        assert.equal(name, targetOrder[index][0] + ' ' + targetOrder[index][1]);
+      });
+    });
+
+    test('Order by family name > ', function() {
+      // Go to settings
+      subject.goToSettings();
+
+      // Change order
+      client.switchToShadowRoot(client.helper.waitForElement(selectors.orderSwitch));
+      subject.clickOn(client.findElement('#switch'));
+      client.switchToShadowRoot();
+      subject.clickOn(client.findElement(selectors.settingsClose));
+      subject.waitForFadeIn(client.helper.waitForElement(selectors.list));
+
+      var names = getNamesInList();
+      var targetOrder = data.sort(familyNameOrder);
+      names.forEach((name, index) => {
+        assert.equal(name, targetOrder[index][0] + ' ' + targetOrder[index][1]);
+      });
+
+    });
```

Comment:
I'm not sure, a `tearDown()` looks missing. What would happen if I added a new test right after this one? Would the order remain by last name?